Model/src/main/java/org/eupathdb/sitesearch/wsfplugin/CommunityStudyIdsPlugin.java

```diff
@@ -0,0 +1,124 @@
+        List<Long> userIds = new ArrayList<>(communityDatasetIds.keySet());
+        Map<Long, User> userMap = userFactory.getUsersById(userIds);
+        for (Long ownerId : userMap.keySet())  {
+            String datasetId = communityDatasetIds.get(ownerId);
+            String ownerName = userMap.get(ownerId).getDisplayName();
+            String institution = userMap.get(ownerId).getOrganization();
+            String[] row = {datasetId, ownerName, institution};
+            response.addRow(row);
+        }
+        return 0;
+    }
+
+    /* we store in memory a map of ownerUserId to VDI dataset id, for each community dataset.
+       We assume there are not too many to fit comfortably into memory.  (100k at absolute most)
+    */
+    protected Map<Long, String> getCommunityDatasetIds(PluginRequest request, Question question, WdkModel wdkModel) throws PluginModelException, PluginUserException {
+
+        if (! wdkModel.getProperties().containsKey(VDI_SCHEMA_SUFFIX_PROP_KEY))
+            throw new PluginModelException("Can't find property'" + VDI_SCHEMA_SUFFIX_PROP_KEY + "' in model.prop file");
+
+        String vdiSchemaSuffix = wdkModel.getProperties().get(VDI_SCHEMA_SUFFIX_PROP_KEY);
+
+        String[] projectsPropList = question.getPropertyList(PROJECT_ID_PROPLIST);
+        if (projectsPropList == null)
+            throw new PluginModelException("Can't find <propertyList> '" + PROJECT_ID_PROPLIST + "' in ID question for community datasets");
+        if (projectsPropList.length != 1)
+            throw new PluginModelException("Error: require a single value in <propertyList> '" + PROJECT_ID_PROPLIST + "' in ID question for community datasets");
+        String projectId = projectsPropList[0];
+
+        DataSource appDs = wdkModel.getAppDb().getDataSource();
+        String sql = "select distinct user_dataset_id, owner " +
+                "from vdi_control_" + vdiSchemaSuffix + ".publicUserDatasets " +
+                "where project_id = '" + projectId + "'";
+        try {
+            return new SQLRunner(appDs, sql).executeQuery(rs -> {
+                Map<Long, String> ownerDatasetMap = new HashMap<>();
+                while (rs.next()) {
+                    String datasetId = rs.getString(1);
+                    Long ownerUserId = rs.getLong(2);
+                    ownerDatasetMap.put(ownerUserId, datasetId);
+                }
+                return ownerDatasetMap;
+            });
+        }
+        catch (SQLRunnerException e) {
+            throw new PluginModelException("Unable to generate project ID map for organism doc type", e.getCause());
+        }
+    }
+
+    @Override
+    public void initialize(PluginRequest request) throws PluginModelException { }
+
+    @Override
+    public void validateParameters(PluginRequest request) throws PluginModelException, PluginUserException { }
+
+    static Question getQuestion(PluginRequest request) throws PluginModelException {
```

Comment:
Remove this and use PluginUtilities.getContextQuestion(request).  May need to update EbrcWebSvcCommon since I just added it (already had a RecordClass lookup so very easy add).

Reply:
i am not currently depending on ebrc code, so i'm gonna leave this as is

Reply:
We could probably move that into WDK... but not today. :)